contracts/strategies/aura/base/AuraStrategyAuraBAL.sol

```diff
@@ -0,0 +1,624 @@
+      );
+    }
+
+    uint256 rewardBalance = IERC20(universalRewardToken).balanceOf(address(this));
+    notifyProfitInRewardToken(rewardBalance);
+    uint256 remainingRewardBalance = IERC20(universalRewardToken).balanceOf(address(this));
+
+    if (remainingRewardBalance == 0) {
+      return;
+    }
+
+    _convertCrvToAuraBAL();
+  }
+
+  function _liquidateBbaUSD()
+    internal
+  {
+    address payable recipient = address(uint160(address(this)));
+    uint256 batchSwapTokens = swapNTokens();
+    uint256 tokenBalance = IERC20(bbaUSD).balanceOf(address(this));
+
+    IERC20(bbaUSD).safeApprove(bVault, 0);
+    IERC20(bbaUSD).safeApprove(bVault, tokenBalance);
+
+    IBVault.BatchSwapStep[] memory swaps = new IBVault.BatchSwapStep[](batchSwapTokens.sub(1));
+    swaps[0].amount = tokenBalance;
+    for (uint256 i = 0; i < batchSwapTokens.sub(1); i++) {
+      swaps[i].poolId = swapPoolIds[i];
+      swaps[i].assetInIndex = i;
+      swaps[i].assetOutIndex = i.add(1);
+    }
+
+    IAsset[] memory assets = new IAsset[](batchSwapTokens);
+    for (uint256 i = 0; i < batchSwapTokens; i++) {
+      assets[i] = IAsset(swapAssets[i]);
+    }
+    
+    IBVault.FundManagement memory funds;
+    funds.sender = address(this);
+    funds.recipient = recipient;
+
+    int256[] memory limits = new int256[](batchSwapTokens);
+    limits[0] = int256(tokenBalance);
+
+    IBVault(bVault).batchSwap(
+      IBVault.SwapKind.GIVEN_IN,
+      swaps,
+      assets,
+      funds,
+      limits,
+      999999999999999999
+    );
+  }
+
+  // Deposit WETH to Balancer and Receive 80BAL20WETH BPT
+  // Deposit 80BAL20WETH to Aura: auraBALBpt Depositor and Receive auraBALBpt
+  function _convertCrvToAuraBAL() 
+    internal 
+  {
+    address universalRewardToken = rewardToken();
+    uint256 depositTokensAmount = depositNTokens();
+    uint256 tokenBalance = IERC20(universalRewardToken).balanceOf(address(this));
+
+    IERC20(universalRewardToken).safeApprove(bVault, 0);
+    IERC20(universalRewardToken).safeApprove(bVault, tokenBalance);
+
+    // we can accept 0 as minimum, this will be called only by trusted roles
+    
+    IAsset[] memory assets = new IAsset[](depositTokensAmount);
+    for (uint256 i = 0; i < depositTokensAmount; i++) {
+      assets[i] = IAsset(poolAssets[i]);
+    }
+
+    IBVault.JoinKind joinKind = IBVault.JoinKind.EXACT_TOKENS_IN_FOR_BPT_OUT;
+    uint256[] memory amountsIn = new uint256[](depositTokensAmount);
+    amountsIn[depositArrayIndex()] = tokenBalance;
+    uint256 minAmountOut = 1;
+
+    bytes memory userData = abi.encode(joinKind, amountsIn, minAmountOut);
+
+    IBVault.JoinPoolRequest memory request;
+    request.assets = assets;
+    request.maxAmountsIn = amountsIn;
+    request.userData = userData;
+    request.fromInternalBalance = false;
+
+    IBVault(bVault).joinPool(
+      balancerPoolId(),
+      address(this),
+      address(this),
+      request
+    );
+
```

Comment:
Would it be possible to implement something like this: https://github.com/harvest-finance/harvest-strategy/pull/71/files#diff-e96312f746bad02267da18daa9abef9d0f1b71c078c0888c02a4df5da415ba6bR254, here too. So check if we can swap the depositToken to the underlying for a better rate than depositing 1:1? Because almost always you get a 2-3% better result when buying auraBal in the market instead of depositing.